src/utils/formatNumbers.ts

```diff
@@ -300,40 +358,65 @@ const TYPE_TO_FORMATTER_RULES = {
   [NumberType.NFTCollectionStats]: ntfCollectionStatsFormatter,
 }
 
-function getFormatterRule(input: number, type: NumberType): Format {
+function getFormatterRule(input: number, type: NumberType): FormatterRule {
   const rules = TYPE_TO_FORMATTER_RULES[type]
   for (const rule of rules) {
     if (
       (rule.exact !== undefined && input === rule.exact) ||
       (rule.upperBound !== undefined && input < rule.upperBound)
     ) {
-      return rule.formatter
+      return rule
     }
   }
 
   throw new Error(`formatter for type ${type} not configured correctly`)
 }
 
-export function formatNumber(
-  input: Nullish<number>,
-  type: NumberType = NumberType.TokenNonTx,
-  placeholder = '-'
-): string {
+interface FormatNumberOptions {
+  input: Nullish<number>
+  type?: NumberType
+  placeholder?: string
+  locale?: SupportedLocale
+  localCurrency?: SupportedLocalCurrency
+}
+
+export function formatNumber({
+  input,
+  type = NumberType.TokenNonTx,
+  placeholder = '-',
+  locale = DEFAULT_LOCALE,
+  localCurrency = DEFAULT_LOCAL_CURRENCY,
+}: FormatNumberOptions): string {
   if (input === null || input === undefined) {
     return placeholder
   }
 
-  const formatter = getFormatterRule(input, type)
-  if (typeof formatter === 'string') return formatter
-  return formatter.format(input)
+  const { hardCodedInput, formatterOptions } = getFormatterRule(input, type)
+
+  if (formatterOptions.currency) {
+    formatterOptions.currency = localCurrency
+    formatterOptions.currencyDisplay = LOCAL_CURRENCY_SYMBOL_DISPLAY_TYPE[localCurrency]
+  }
+
+  if (!hardCodedInput) {
+    return new Intl.NumberFormat(locale, formatterOptions).format(input)
+  }
```

Comment:
are you introducing the concept of "hardcoded" input/output values in this PR? can you add a comment somewhere explaining why it's needed?

Reply:
i am not really introducing it - it kinda already existed already (I more dissected it out of the current functionality) but I actually explained it above the type definition of it